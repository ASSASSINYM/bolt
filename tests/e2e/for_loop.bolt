void main(){
  for(let i = 0; i< 10 ; i:=i+1){
    printf("i = %d;", i)
  }
}