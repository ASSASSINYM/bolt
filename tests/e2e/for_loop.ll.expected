; ModuleID = 'Module'
source_filename = "Module"
target triple = "x86_64-apple-darwin18.7.0"

%pthread_t = type opaque

@0 = private unnamed_addr constant [9 x i8] c"i = %d;\0A\00", align 1

declare i32 @printf(i8*, ...)

declare i8* @GC_malloc(i64)

declare i32 @pthread_create(%pthread_t**, i8*, i8* (i8*)*, i8*)

declare i32 @pthread_join(%pthread_t*, i8**)

declare i32 @pthread_equal(%pthread_t*, %pthread_t*)

declare %pthread_t* @pthread_self()

define i32 @main() {
entry:
  br label %loop

loop:                                             ; preds = %entry, %loop
  %_i0.0 = phi i32 [ 0, %entry ], [ %add, %loop ]
  %0 = call i32 (i8*, ...) @printf(i8* getelementptr inbounds ([9 x i8], [9 x i8]* @0, i64 0, i64 0), i32 %_i0.0)
  %add = add i32 %_i0.0, 1
  %lt1 = icmp slt i32 %add, 10
  br i1 %lt1, label %loop, label %loopend

loopend:                                          ; preds = %loop
  ret i32 0
}
